frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -347,22 +347,18 @@ pub mod pallet {
         ///
         /// Sudo call is required
         #[pallet::weight(1_000_000)]
-        pub fn contract_pre_approval(
+        pub fn developer_pre_approval(
             origin: OriginFor<T>,
-            contract_id: SmartContract<T::AccountId>,
+            developer: T::AccountId,
         ) -> DispatchResultWithPostInfo {
             ensure_root(origin)?;
-            ensure!(
-                !RegisteredDapps::<T>::contains_key(&contract_id),
-                Error::<T>::AlreadyRegisteredContract
-            );
 
-            let pre_approved_contracts = Self::pre_approved_contracts();
+            let pre_approved_developers = Self::pre_approved_developers();
             ensure!(
-                !pre_approved_contracts.contains(&contract_id),
-                Error::<T>::AlreadyPreApprovedContract
+                !pre_approved_developers.contains(&developer),
+                Error::<T>::AlreadyPreApprovedDeveloper
```

Comment:
Didn't this get commented in another PR before?

That you should use a map instead of a vector?

Reply:
what would be the key/value for that map?
during the `developer_pre_approval()`call we only know developer's accountId

Reply:
I think we commented that too but there are many PRs that are closed now.

You can have a map that maps accountId => ().
That will make it act as a set and you can check if a key exists without reading the whole structure in.

Reply:
implemented StorageMap(accountId => ())